staging/src/k8s.io/apiserver/pkg/apis/config/types.go

```diff
@@ -89,4 +89,22 @@ type KMSConfiguration struct {
 	// Timeout for gRPC calls to kms-plugin (ex. 5s). The default is 3 seconds.
 	// +optional
 	Timeout *metav1.Duration
+	// HealthzEndpoint contains information required to validate healthz of kms-plugin.
+	// +optional
+	HealthzEndpoint *HealthzEndpoint
+}
+
+// HealthzEndpoint contains information required to validate healthz of kms-plugin.
+type HealthzEndpoint struct {
+	// URL is the url on which kube-apiserver could check the health status of kms-plugin.
+	URL string
+	// Timeout defines when to give-up on sending healthz probes (defaults to 90 seconds).
+	// +optional
+	Timeout *metav1.Duration
+	// RequestTimeout defines timeout for an individual healthz request (defaults to 3 seconds).
+	// +optional
+	RequestTimeout *metav1.Duration
```

Comment:
without success/failure thresholds, does that mean a single failure here can put the apiserver in an unhealthy state briefly?

Reply:
A single 500 status from kms-plugin's healthz during the initialization of kube-apiserver will trigger an error - preventing start-up of kube-apiserver.
503 and lack of response will be retried for the duration of the timeout.